src/SonarScanner.MSBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -96,18 +108,27 @@ public IAnalyzerProvider CreateRoslynAnalyzerProvider(ISonarWebServer server, st
             return new RoslynAnalyzerProvider(new EmbeddedAnalyzerInstaller(server, localCacheTempPath, logger), logger);
         }
 
-        private async Task<Version> QueryServerVersion(IDownloader downloader)
+        private async Task<Version> QueryServerVersion(IDownloader downloader, IDownloader fallback)
         {
             logger.LogDebug(Resources.MSG_FetchingVersion);
+
             try
             {
-                var contents = await downloader.Download("api/server/version");
-                return new Version(contents.Split('-').First());
+                var contents = await downloader.Download("analysis/version");
+                return new Version(contents.Split('-')[0]);
```

Comment:
To avoid code duplication you can extract
```csharp
        private static async Task<Version> GetVersion(IDownloader downloader, string path)
        {
            var contents = await downloader.Download(path);
            return new Version(contents.Split('-')[0]);
        }
```